src/components/RestaurantItemBottomSheet.ts

```diff
@@ -0,0 +1,108 @@
+import Bookmark from "./common/Bookmark";
+
+class RestaurantItemBottomSheet {
+  restaurant: Restaurant;
+  bookmark: Bookmark;
+  deleteItem: (id: string) => void;
+
+  constructor(restaurant: Restaurant, deleteItem: (id: string) => void) {
+    this.restaurant = restaurant;
+    this.deleteItem = deleteItem;
+    const bookmark = new Bookmark(this.restaurant);
+    this.bookmark = bookmark;
+  }
+
+  template() {
+    return `<div class="item-sheet" data-id="${this.restaurant.id}">
+      <div class="modal-backdrop item-sheet-backdrop"></div>
+      <div class="modal-container">
+      <div class="item__header">
+        <div class="restaurant__category">
+          <img src=${categoryToSrc(<Category>this.restaurant.category)} alt=${
+      this.restaurant.category
+    } class="category-icon">
+            </div>
+            ${this.bookmark.template()}
+        </div>
+            <div class="item__info">
+          <h3 class="item__name text-subtitle">${this.restaurant.name}</h3>
+          <span class="item__distance text-body">캠퍼스부터 ${
+            this.restaurant.takingTime
+          }분 내</span>
+          <p class="item__description text-body">${
+            this.restaurant.description
+          }</p>
+          <p>${this.restaurant.link}</p>
+        </div>
+          <div class="button-container">
+          <button type="submit" class="button button--secondary text-caption item-sheet--delete">삭제하기</button>
+          <button type="button" class="button button--primary text-caption item-sheet--close">닫기</button>
+          </div>
+        </div>
+    </div>`;
+  }
+
+  initialize(rerenderList: () => void) {
+    this.render();
+    this.addEvent(rerenderList);
+  }
+
+  render() {
+    $("body")?.insertAdjacentHTML("beforeend", this.template());
+  }
+
+  addEvent(rerenderList: () => void) {
+    this.handleBookmark(rerenderList);
+    this.handleSheetDelete();
+    this.handleSheetClose();
+  }
+
+  handleBookmark(rerenderList: () => void) {
+    const bottomSheetHeader = <HTMLElement>$(".item__header");
+    bottomSheetHeader?.addEventListener("click", (event) => {
+      const bookmarkButton = <HTMLElement>event.target;
+      const id = this.restaurant.id;
+
+      if (bookmarkButton) {
+        restaurantListHandler.toggleBookmark(id);
+
+        bookmarkButton.remove();
+        this.restaurant.bookmark = !this.restaurant.bookmark;
+        bottomSheetHeader.insertAdjacentHTML(
+          "beforeend",
+          this.bookmark.template()
+        );
+
+        rerenderList();
+        return;
+      }
+    });
+  }
+
+  handleSheetDelete() {
+    $(".item-sheet--delete")?.addEventListener("click", () => {
+      this.deleteItem(this.restaurant.id);
+      this.closeSheet();
+    });
+  }
+
+  handleSheetClose() {
+    $(".item-sheet--close")?.addEventListener("click", () => {
+      this.closeSheet();
+    });
+
+    $(".item-sheet-backdrop")?.addEventListener("click", () => {
+      this.closeSheet();
+    });
```

Comment:
이런 경우 이벤트 위임도 고려해보시면 좋아요!
아니면 아래와 같이도 가능합니다 ㅎㅎ
```suggestion
    const itemSheetElements = [$(".item-sheet--close"), $(".item-sheet-backdrop")];

    itemSheetElements.forEach((itemSheetElement) => itemSheetElement?.addEventListener("click", () => {
      this.closeSheet();
    }))
```

Reply:
우와 요소를 배열에 넣어서 forEach를 하는 방법도 있었군요! 멋져요!! 👍